Library/Homebrew/cmd/--selfdestruct.rb

```diff
@@ -0,0 +1,93 @@
+# Original Inspiration: Stephen Benner https://github.com/SteveBenner: https://gist.github.com/SteveBenner/11254428
+# Partially Rewritten by Dominyk Tiller https://github.com/DomT4 for Homebrew's internal use.
+# THIS WILL COMPLETELY REMOVE HOMEBREW FROM YOUR SYSTEM
+
+require "fileutils"
+require "open3"
+require "rubygems"
+
+begin
+  require "trollop"
+rescue LoadError
+  onoe "You must run `sudo /System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install trollop`"
+  raise
+end
```

Comment:
> This isn't completely without precedence though. Various aspects of Brew will require you to install x gem for x purpose.

But those features aren't for uninstalling something. `ARGV` should be enough in this script.